pages/admin/site/moderation.vue

```diff
@@ -0,0 +1,377 @@
+import { BrandedButton } from '@datagouv/components-next'
+import type { PaginatedArray } from '~/types/types'
+import AdminBreadcrumb from '~/components/Breadcrumbs/AdminBreadcrumb.vue'
+import BreadcrumbItem from '~/components/Breadcrumbs/BreadcrumbItem.vue'
+import AdminTable from '~/components/AdminTable/Table/AdminTable.vue'
+import AdminTableTh from '~/components/AdminTable/Table/AdminTableTh.vue'
+import type { Thread } from '~/types/discussions'
+import DatasetBadge from '~/components/AdminBadge/DatasetBadge.vue'
+import DataserviceBadge from '~/components/AdminBadge/DataserviceBadge.vue'
+import ReuseBadge from '~/components/AdminBadge/ReuseBadge.vue'
+
+const { t } = useTranslation()
+const { formatDate } = useFormatDate()
+const { $api } = useNuxtApp()
+
+const page = ref(1)
+const pageSize = ref(20)
+
+const filterStatus = ref({ label: t('En cours'), id: 'ongoing' })
+const filterStatusValue = computed(() => filterStatus.value.id || undefined)
+
+const { data: pageData, status, refresh } = await useAPI<PaginatedArray<Report>>(`/api/1/reports/`, { query: {
+  status: filterStatusValue,
+} })
+const { data: reasons } = await useAPI<Array<ReportReason>>('/api/1/reports/reasons/', { lazy: true })
+
+const getSubjectUrl = (subject: ReportSubject) => {
+  return {
+    Dataset: `/api/2/datasets/${subject.id}/`,
+    Dataservice: `/api/1/dataservices/${subject.id}/`,
+    Reuse: `/api/1/reuses/${subject.id}/`,
+    Organization: `/api/1/organizations/${subject.id}/`,
+    Discussion: `/api/1/discussions/${subject.id}/`,
+  }[subject.class]
+}
+
+type RecordSubjectFullObject = { type: 'Dataset', value: null | DatasetV2 }
+  | { type: 'Dataservice', value: null | Dataservice }
+  | { type: 'Reuse', value: null | Reuse }
+  | { type: 'Organization', value: null | Organization }
+  | { type: 'Discussion', value: null | Thread }
+
+const subjects = ref({} as Record<string, RecordSubjectFullObject>)
+const fetchFullSubject = async (report: Report, subject: ReportSubject) => {
+  if (report.subject_deleted_at) {
+    subjects.value[report.id] = { type: subject.class, value: null }
+  }
+
+  try {
+    const value = await $api(getSubjectUrl(subject))
+    subjects.value[report.id] = { type: subject.class, value }
+  }
+  catch {
+    subjects.value[report.id] = { type: subject.class, value: null }
+  }
+}
+
+watch(pageData, async () => {
+  console.log('pageData changed')
+  if (!pageData.value) return
+  for (const report of pageData.value.data) {
+    const subject = report.subject
+    if (!subject) continue
+    if (subjects.value[report.id]) continue
+
+    await fetchFullSubject(report, subject)
+  }
+}, { immediate: true })
+
+const activities = ref({} as Record<string, Activity>)
+watchEffect(async () => {
+  if (!pageData.value) return
+
+  const subjects = pageData.value.data.map(r => r.subject as ReportSubject).filter(s => s)
+
+  const newActivities = await getLatestActivitiesForObjects($api, subjects)
```

Comment:
We're fetching the last activity but we're showing it in the column `Créé le`, which make it seems like it's the creation activity.

Reply:
There is 2 different date
- The objet creation date
- The trigger date

I'm not sure I understand